whisper/whisperv5/peer.go

```diff
@@ -34,20 +34,23 @@ type Peer struct {
 	ws      p2p.MsgReadWriter
 	trusted bool
 
-	known *set.Set // Messages already known by the peer to avoid wasting bandwidth
-
-	quit chan struct{}
+	known      *set.Set // Messages already known by the peer to avoid wasting bandwidth
+	advertised map[common.Hash]struct{}
```

Comment:
would it be possible to have `advertised` as a global table? If new nodes connect, they would not receive older messages, but it would save a lot of memory; and it's consistent with the behavior that filters will miss messages that were sent before those filters were installed.

In any case, I'd be curious to find out what the results are, in terms of duplicates, when the table is global.